File: Data Structures & Algorithms/max-water-container/submission-2.cpp
class Solution {
public:
    int maxArea(vector<int>& heights) {
        int left = 0, right = heights.size() - 1;
        int res = 0;
        while(left < right){
            res = max(res, min(heights[left], heights[right]) * (right-left));

            if(heights[left] < heights[right]){
                left++;
            } else{
                right--;
            }
        }
        return res;
    }
};

File: Data Structures & Algorithms/eating-bananas/submission-2.py
import math
class Solution:
    def minEatingSpeed(self, piles: List[int], h: int) -> int:
        Min, Max = 1, 0
        for cnt in piles:
            if cnt < Min:
                Min = cnt
            if cnt > Max:
                Max = cnt
        res = float('inf')
        print(Min, Max)
        while Min <= Max:
            k = (Min + Max) // 2
            eatingTime = 0
            for cnt in piles:
                eatingTime += math.ceil(cnt / k)
            if eatingTime > h:
                Min = k + 1
            elif eatingTime <= h:
                Max = k - 1
                
            if eatingTime <= h:
                res = min(k, res)
            print(eatingTime, k)
        return int(res)